service/s3/doc.go

```diff
@@ -17,6 +17,9 @@
 // See the SDK's documentation for more information on how to use the SDK.
 // https://docs.aws.amazon.com/sdk-for-go/api/
 //
+// Interacting with objects in an empty prefix (e.g. /bucketname/foo//bar/objectname)
```

Comment:
I think we'll also want to highlight this a little bit more genericly for object keys with adjacent slashes, dot slash (`./`) and  should go into the `doc_custom.go` file.

Maybe something similar to solution section added to #1853